File: C++ 알고리즘/EPPER/입국심사/입국심사/입국심사.cpp
#include <vector>
#include <algorithm>
#include <climits>

using namespace std;

long long solution(int n, vector<int> times) {
	long long ans = LLONG_MAX;
	sort(times.begin(), times.end());
	long long start = 1;
	long long end = (long long)n * (long long)times[times.size() - 1];
	long long mid = 0;
	while (start <= end) {
		int temp = 0;
		mid = (start + end) / 2;
		for (int i = 0; i < times.size(); i++) {
			temp += mid / times[i];
		}
		if (temp >= n) {
			ans = min(ans, mid);
			end = mid - 1;
		}
		else start = mid + 1;
	}
	return ans;
}

File: 프로그래머스/입국심사.js
function solution(n, times) {
    let answer=0;
    times.sort((a, b)=> b-a);
    let start=1, end = times[0]*n; // 최악의 경우 계산
    
    while(start<=end){
        let mid = Math.floor((start+end)/2); // 초
        
        let total = 0;
        for(const time of times){ // total초 동안 각 심사대는 Math.floor(mid/time) 명 심사 가능
            total+=Math.floor(mid/time);
        }
        
        if(total>=n) { // 다 심사할 수 있는 경우 -> 심사시간 줄이기
            answer = mid;
            end = mid-1;
        }
        else { // 다 심사하지 못하는 경우 -> 심사시간 늘이기
            start = mid+1; 
        }
    }
    return answer;
}